Add new LMPOP and BLMPOP commands. (#1636)
redis/redis-doc: commands/lmpop.md

Pops one ore more elements from the first non-empty list key from the list of provided key names.

> Similar other commands have limitations:

Reviewer: ```suggestion
LMPOP and BLMPOP deprecate the following commands:
```

Reviewer: @enjoy-binbin removed the deprecation mention after we agreed we're not deprecating the simple commands. Please review the suggestion above.

Author: Took it! thanks for the all effort

> - `LPOP` or `RPOP` takes one key, and can return multiple elements.

Reviewer: ```suggestion
- `LPOP` or `RPOP` which take only one key, and can return multiple elements.
```

> `LPOP` or `RPOP` takes one key, and can return multiple elements.
> `BLPOP` or `BRPOP` takes multiple keys, but returns one element from just one key.
> `LMPOP` or `BLMPOP` can take multiple keys and return multiple elements from just one key.

Reviewer 2: ```suggestion
Pops one ore more elements from the first non-empty list key from the list of provided key names.

Similar other commands have limitations:
- `LPOP` or `RPOP` takes one key, and can return multiple elements.
- `BLPOP` or `BRPOP` takes multiple keys, but returns one element from just one key.

See `BLMPOP` for the blocking variant of this command.
```

> - `BLPOP` or `BRPOP` takes multiple keys, but returns one element from just one key.

Reviewer: ```suggestion
- `BLPOP` or `BRPOP` which take multiple keys, but return only one element from just one key.
```

See `BLMPOP` for the blocking variant of this command.

Search the list from left to right, find the first non-empty list, return the corresponding list key name, pop from the left or right depending on the argument given, and return the elements.

> The number of returned elements will be limited to `count`(default 1) and the list length.

Reviewer: Elements are popped from either the left or right of the first non-empty list based on the `where` argument. Number of returned elements is limited to the lower between the non-empty list's length and the `count` argument (which defaults to 1).

Author: About the `where` arg
Oran said this in the conversation above: I don't think we wanna refer to where IIRC the argument name isn't shown.

Reviewer: Elements are popped from either the left or right of the first non-empty list based on the passed argument. The number of returned elements is limited to the lower between the non-empty list's length and the `count` argument (which defaults to 1).

@return

@array-reply: specifically:

> * A `nil` multi-bulk when no element could be popped.
> * A two-element multi-bulk with the first element being the name of the key where
>   elements was popped, and the second element is a multi-bulk with an array of elements.

Reviewer 2: ```suggestion
* A `nil` when no element could be popped.
* A two-element array with the first element being the name of the key from which elements where popped, and the second element is an array of elements.
```

> * A two-element array with the first element being the name of the key from which elements where popped, and the second element is an array of elements.

Reviewer: ```suggestion
* A two-element array with the first element being the name of the key from which elements were popped, and the second element is an array of elements.
```
